core/src/serde_support.rs

```diff
@@ -0,0 +1,1458 @@
+
+use alloc::format;
+use alloc::string::ToString;
+use alloc::vec::Vec;
+
+use base64 as _;
+use base64::engine::general_purpose::STANDARD as BASE64_STANDARD;
+use base64::Engine;
+use dusk_bytes::Serializable;
+use hex as _;
+use serde::de::{
+    EnumAccess, Error as SerdeError, MapAccess, Unexpected, VariantAccess,
+    Visitor,
+};
+use serde::ser::SerializeStruct;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_json as _;
+
+use crate::stake::{Reward, RewardReason, SlashEvent, StakeEvent, StakeKeys};
+use crate::transfer::withdraw::WithdrawReceiver;
+use crate::transfer::{
+    ContractToAccountEvent, ContractToContractEvent, ConvertEvent,
+    DepositEvent, MoonlightTransactionEvent, PhoenixTransactionEvent,
+    WithdrawEvent,
+};
+use crate::{
+    signatures::bls::PublicKey as AccountPublicKey, stake::StakeFundOwner,
+};
+use crate::{BlsScalar, String};
+
+// To serialize and deserialize u64s as big ints:
+// https://github.com/dusk-network/rusk/issues/2773#issuecomment-2519791322.
+struct Bigint(u64);
+
+impl Serialize for Bigint {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s: String = format!("{}n", self.0);
+        s.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for Bigint {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let mut s = String::deserialize(deserializer)?;
+        let last_char = s.pop().ok_or_else(|| {
+            SerdeError::invalid_value(
+                Unexpected::Str(&s),
+                &"a non-empty string",
+            )
+        })?;
+        if last_char != 'n' {
+            return Err(SerdeError::invalid_value(
+                Unexpected::Str(&s),
+                &"a bigint ending with character 'n'",
+            ));
+        }
+        let parsed_number = u64::from_str_radix(&s, 10).map_err(|e| {
+            SerdeError::custom(format!("failed to deserialize u64: {e}"))
+        })?;
+        Ok(Self(parsed_number))
+    }
+}
+
+impl Serialize for StakeFundOwner {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        match self {
+            Self::Account(pk) => serializer.serialize_newtype_variant(
+                "StakeFundOwner",
+                0,
+                "Account",
+                pk,
+            ),
+            Self::Contract(contract_id) => serializer
+                .serialize_newtype_variant(
+                    "StakeFundOwner",
+                    1,
+                    "ContractId",
+                    contract_id,
+                ),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for StakeFundOwner {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        struct StakeFundOwnerVisitor;
+
+        static VARIANTS: [&'static str; 2] = ["Account", "ContractId"];
```

Comment:
```suggestion
        const VARIANTS: [&str; 2] = ["Account", "ContractId"];
```

any reason to have this as static?

Reply:
Actually, no.